src/Files.App/Helpers/Layout/LayoutPreferencesManager.cs

```diff
@@ -40,121 +40,115 @@ public bool IsAdaptiveLayoutEnabled
+					FolderLayoutModes.ColumnView => LayoutPreferencesItem.IconSizeColumnsView,
+					_ => Constants.Browser.GridViewBrowser.GridViewSizeMedium,
+				};
+			}
 			set
 			{
-				// Size down
-				if (value < LayoutPreferencesItem.IconSizeGridView)
+				switch (LayoutMode)
 				{
-					// Size down from List to Details
-					if (LayoutMode == FolderLayoutModes.ListView)
-					{
-						LayoutPreferencesItem.IsAdaptiveLayoutOverridden = true;
-						LayoutMode = FolderLayoutModes.DetailsView;
-						LayoutModeChangeRequested?.Invoke(this, new LayoutModeEventArgs(LayoutMode, GridViewSize));
-					}
-					// Size down from Tiles to List
-					else if (LayoutMode == FolderLayoutModes.TilesView)
-					{
-						LayoutPreferencesItem.IsAdaptiveLayoutOverridden = true;
-						LayoutMode = FolderLayoutModes.ListView;
-						LayoutModeChangeRequested?.Invoke(this, new LayoutModeEventArgs(LayoutMode, GridViewSize));
-					}
-					// Size down from Grid to Tiles
-					else if (LayoutMode == FolderLayoutModes.GridView && value < Constants.Browser.GridViewBrowser.GridViewSizeSmall)
-					{
-						LayoutPreferencesItem.IsAdaptiveLayoutOverridden = true;
-						LayoutMode = FolderLayoutModes.TilesView;
-						LayoutModeChangeRequested?.Invoke(this, new LayoutModeEventArgs(LayoutMode, GridViewSize));
-					}
-					// Resize grid view
-					else if (LayoutMode != FolderLayoutModes.DetailsView)
-					{
-						// Set grid size to allow immediate UI update
-						var newValue = (value >= Constants.Browser.GridViewBrowser.GridViewSizeSmall) ? value : Constants.Browser.GridViewBrowser.GridViewSizeSmall;
-						SetProperty(ref LayoutPreferencesItem.IconSizeGridView, newValue, nameof(GridViewSize));
-
-						// Only update layout mode if it isn't already in grid view
-						if (LayoutMode != FolderLayoutModes.GridView)
-						{
-							LayoutPreferencesItem.IsAdaptiveLayoutOverridden = true;
-							LayoutMode = FolderLayoutModes.GridView;
-							LayoutModeChangeRequested?.Invoke(this, new LayoutModeEventArgs(LayoutMode, GridViewSize));
-						}
-						else
-						{
-							LayoutPreferencesUpdateRequired?.Invoke(this, new LayoutPreferenceEventArgs(LayoutPreferencesItem));
-						}
-
-						GridViewSizeChangeRequested?.Invoke(this, EventArgs.Empty);
-					}
+					case FolderLayoutModes.DetailsView:
+						SetProperty(ref LayoutPreferencesItem.IconSizeDetailsView, value, nameof(IconSize));
+						break;
+					case FolderLayoutModes.ListView:
+						SetProperty(ref LayoutPreferencesItem.IconSizeListView, value, nameof(IconSize));
+						break;
+					case FolderLayoutModes.TilesView:
+						SetProperty(ref LayoutPreferencesItem.IconSizeTilesView, value, nameof(IconSize));
+						break;
+					case FolderLayoutModes.GridView:
+						SetProperty(ref LayoutPreferencesItem.IconSizeGridView, value, nameof(IconSize));
+						break;
+					case FolderLayoutModes.ColumnView:
+						SetProperty(ref LayoutPreferencesItem.IconSizeColumnsView, value, nameof(IconSize));
+						break;
```

Comment:
You could convert this to switch expression.

Reply:
Let's leave as is for now